src/loaders/OmeZarrLoader.ts

```diff
@@ -112,6 +114,35 @@ const DEFAULT_FETCH_OPTIONS = {
   maxPrefetchChunks: 30,
 };
 
+function getOMEMetadata(group: zarr.Group<Store>): OMEZarrMetadata {
+  const attrs = group.attrs;
+  if (!attrs.multiscales) {
+    // try for another field
+    if (attrs.ome) {
+      const ome = attrs.ome as any;
+      if (ome.version) {
+        if (ome.version !== "0.5") {
+          throw new VolumeLoadError(`OME-Zarr metadata version ${ome.version} not supported`, {
+            type: VolumeLoadErrorType.INVALID_METADATA,
+          });
+        }
+      }
+      if (ome.multiscales) {
+        return { multiscales: ome.multiscales as OMEMultiscale[], omero: ome.omero as OmeroTransitionalMetadata };
+      } else {
+        throw new VolumeLoadError("OME-Zarr metadata missing 'multiscales' field", {
+          type: VolumeLoadErrorType.INVALID_METADATA,
+        });
+      }
+    } else {
+      throw new VolumeLoadError("OME-Zarr metadata missing 'multiscales' or 'ome' field", {
+        type: VolumeLoadErrorType.INVALID_METADATA,
+      });
+    }
+  }
+  return { multiscales: attrs.multiscales as OMEMultiscale[], omero: attrs.omero as OmeroTransitionalMetadata };
+}
```

Comment:
fwiw a lot of the validation this function does is already done by `validateOMEZarrMetadata` below. This could probably be replaced at the callsite with something like `const metadata = group.attrs.ome ?? group.attrs;` plus the version check (which could also go in `validateOMEZarrMetadata`).

Reply:
Thinking ahead to future versions, I thought it would be good to isolate this in its own function.  And if the schemas diverge even more, we may have to read the data into our own internal structure.  But for now, maybe it can be combined with the validate function somehow?

Reply:
I definitely like the idea of keeping any sort of parsing separate. My thought was that long-term, if we want to do any more complex parsing, that parsing would go in the current validation module, which would get renamed to reflect its new responsibilities. (In a sense, validation is just a special case of parsing where we don't do any transformation.) That could also happen now though. The current type of `validateOMEZarrMetadata` is `(data: unknown, ...) => asserts data is OMEZarrMetadata` - i.e. returns nothing, but if control flow passes through it safely then `data` has been validated. If it's also responsible for that simple transformation, it can just as easily guarantee that you have properly formatted metadata by passing it back like `(data: unknown, ...) => OMEZarrMetadata`.